week1/armstrong.rb

```diff
@@ -7,7 +7,41 @@
 #
 # Your output should be `true` if the number is an Armstrong number
 # and `false` otherwise.
-
-def armstrong_number(n)
-    raise NotImplementedError # Delete this line and start coding here
-end
+def pow(a,b)
+    result = 1
+    b.times do
+        result=result*a
+    end
+    return result
+  end
+  def armstrong_number(n)
+    total_digits=0
+    digit=0
+    num=n
+    while(num>0)
+      #digit=num%10
+      num/=10
+      total_digits=total_digits+1
+    end
+    i=1
+    sum=0
+    number=n
+    while i<=total_digits
+      digit=n%10
+      n=n/10
+      sum+=pow(digit,total_digits)
+      i+=1
+    end
+  if (number==sum)
+      return "true"
+  else
+      return "false"
+  end
+  
+  end
+  
+  n=gets.chomp().to_i
```

Comment:
Our testing script directly invokes the function. This is unnecessary and also the reason for the check failing